app/api/csv.py

```diff
@@ -142,3 +175,376 @@ def row(r):
+            answers.update({'Response criteria': {}})
+            answers.update({'Optional response criteria': {}})
+            answers.update({'Work eligibility': ''})
+            answers.update({'Previously worked for agency': ''})
+            answers.update({'Meets clearance requirements': ''})
+
+        elif r['slug'] == 'atm':
+            answers.update({'Candidate name': ''})
+            answers.update({'Availability': r['availability']})
+            answers.update({'Daily rate': ''})
+            answers.update({'Hourly rate': ''})
+            answers.update({'Contact number': r['respondToPhone']})
+            answers.update({'Contact email': ''})
+            answers.update({'Response criteria': r['criteria']})
+            answers.update({'Optional response criteria': {}})
+            answers.update({'Work eligibility': ''})
+            answers.update({'Previously worked for agency': ''})
+            answers.update({'Meets clearance requirements': ''})
+
+        elif r['slug'] == 'training2':
+            answers.update({'Candidate name': ''})
+            answers.update({'Availability': ''})
+            answers.update({'Daily rate': ''})
+            answers.update({'Hourly rate': ''})
+            answers.update({'Contact number': r['respondToPhone']})
+            answers.update({'Contact email': r['respondToEmailAddress']})
+            answers.update({'Response criteria': {}})
+            answers.update({'Optional response criteria': {}})
+            answers.update({'Work eligibility': ''})
+            answers.update({'Previously worked for agency': ''})
+            answers.update({'Meets clearance requirements': ''})
+        answers.update({'Responded at': r['created_at']})
+
+        answers.update({'Response criteria': get_response_criteria_array(answers['Response criteria'])})
+        answers.update({
+            'Optional response criteria': get_response_criteria_array(answers['Optional response criteria'])
+        })
+
+        for k, v in answers.items():
+            answers[k] = csv_cell_sanitize(v)
+
+        return answers
+
+    # convert the responses to a list - each row is a dict representing a brief response
+    rows = [row(_) for _ in sellerResponses]
+    # add the keys of a dict as the first element in the rows list, to be used as headers
+    first = rows[0].keys() if sellerResponses else []
+    rows = [first] + [_.values() for _ in rows]
+    # convert the rows to a zip'd list - this creates a structure so the csv's first column is the headers and the
+    # remaining columns contain the data for each header row
+    csvdata = StringIO()
+    with csvx.Writer(csvdata) as csv_out:
+        csv_out.write_rows(rows)
+        return csvdata.getvalue()
+
+
+def generate_specialist_opportunities_csv(specialistOpportunities):
+    # converts a brief response into an ordered dict
+    def row(r):
+        answers = od()
+        answers.update({'opportunity_id': r['id']})
```

Comment:
Do these keys need to be formatted like the ones above? Uppercased with underscores removed?

Reply:
created new function to auto convert if there are underscores in the key